ompi/mca/hook/demo/hook_demo_fns.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016      IBM Corporation.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "hook_demo.h"
+
+#define DEBUG_OUTPUT( fn_name )                     \
+    do {                                            \
+        opal_output(0, "hook:demo: " # fn_name );   \
+    } while(0)
+
+void ompi_hook_demo_mpi_initialized_top(int *flag)
+{
+    DEBUG_OUTPUT( mpi_initialized_top );
+}
+
+void ompi_hook_demo_mpi_initialized_bottom(int *flag)
+{
+    DEBUG_OUTPUT( mpi_initialized_bottom );
+}
+
+
+void ompi_hook_demo_mpi_finalized_top(int *flag)
+{
+    DEBUG_OUTPUT( mpi_finalized_top );
+}
+
+void ompi_hook_demo_mpi_finalized_bottom(int *flag)
+{
+    DEBUG_OUTPUT( mpi_finalized_bottom );
+}
+
+
+void ompi_hook_demo_mpi_init_top(int argc, char **argv, int requested, int *provided)
+{
+    DEBUG_OUTPUT( mpi_init_top );
+}
+
+void ompi_hook_demo_mpi_init_top_post_opal(int argc, char **argv, int requested, int *provided)
+{
+    DEBUG_OUTPUT( mpi_init_top_post_opal );
+}
+
+void ompi_hook_demo_mpi_init_bottom(int argc, char **argv, int requested, int *provided)
+{
+    DEBUG_OUTPUT( mpi_init_bottom );
+}
+
+void ompi_hook_demo_mpi_init_error(int argc, char **argv, int requested, int *provided)
+{
+    DEBUG_OUTPUT( mpi_init_error );
+}
+
+
+void ompi_hook_demo_mpi_finalize_top(void)
+{
+    DEBUG_OUTPUT( mpi_finalize_top );
+}
+
+void ompi_hook_demo_mpi_finalize_bottom(void)
+{
+    DEBUG_OUTPUT( mpi_finalize_bottom );
+}
+
+
+void ompi_hook_demo_extra_mpi_init_bottom(int argc, char **argv, int requested, int *provided)
+{
+    DEBUG_OUTPUT( mpi_init_bottom (extra) );
```

Comment:
I could not find where `mpi_init_bottom` was defined, nor `extra`..., but yet this file still compiled for me.  Whaaa...?  I have to run now and can't investigate further...